### ISSUE TITLE
Lint/NonLocalExitFromIterator false positive

### ISSUE THREAD

**Issue by zachlatta** [bbatsov/rubocop#3855]
I think I'm get a false positive from `Lint/NonLocalExitFromIterator` in the below code:

```rb 
VCR.configure do |c|
  # ...

  def server_cookies(response)
    raw_cookies = response.headers['Set-Cookie']
    return if raw_cookies.nil? || raw_cookies.empty?

    raw_cookies.map { |ck| parse_server_cookie(ck) }
  end
end
```

--------

## Expected behavior

No flags should be raised.

## Actual behavior

It's flagged with:

```
W: Non-local exit from iterator, without return value. next, break, Array#find, Array#any?, etc. is preferred.
```

Curiously, this only seems to happen when there's a block argument specified in the `VCR.configure` block – try omitting `|c|` to see what I mean.

## Steps to reproduce the problem

1. Create a file with the above code snippet as its contents
2. Run Rubocop on it

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
bundle exec rubocop -V
0.46.0 (using Parser 2.3.3.1, running on ruby 2.3.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
Nice find @zachlatta! I think the reason we've missed this is that it's a pretty uncommon (at least to me) edge case. But there's no reason why we couldn't account for this, that I can see. 😀

**Comment by bbatsov** [comment]
Yeah, that should be easy to fix.
